### PR TITLE
Rewrite in python

### PR DESCRIPTION
**Pull request by eroussy**
This PR rewrite the compile.sh script in python.

From a user perspective, there is two visible changes : 
- The `-c` option to select a compliance matrix file is now usable multiple times in order to create multiple matrix
- There is a new option `-m` to add the machine name after each test table name in the pdf. This machine name is the classname of the first test in the suite.

### CODE REVIEW

File: .cqfd/docker/Dockerfile [hunk 1]
```diff
@@ -16,18 +16,41 @@ FROM ubuntu:22.04
...
+    && unzip /tmp/sonar-scanner.zip \
+    && rm -f sonar-scanner.zip
+RUN ln -s "/opt/sonar-scanner-${sonar_version}" /opt/sonar-scanner
+RUN echo 'sonar.host.url=http://j1.sfl.team:9000/' \
+    > /opt/sonar-scanner/conf/sonar-scanner.properties
+COPY python-sonar.sh /usr/bin/python-sonar.sh
+COPY detect_sonar_errors.py /usr/bin/detect_sonar_errors.py
+
+# Install cukinia to compile examples
+RUN git clone https://github.com/savoirfairelinux/cukinia.git /tmp/cukinia
```
Comment (dupremathieu) [inline .cqfd/docker/Dockerfile:55]: Replace it by:
ADD https://raw.githubusercontent.com/savoirfairelinux/cukinia/master/cukinia /usr/bin/cukinia
RUN chmod +x /usr/bin/cukinia

You can remove the git package.

File: compile.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
+#!/bin/python3
+#
+# Copyright (C) 2023 Savoir-faire Linux, Inc.
```
Comment (dupremathieu) [inline compile.py:3]: Who has chosen the license?
In the other project, we have dual GPLv3 and Apache license.

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+sonar.report.export.path=sonar-report.json
+# Only sonar for Python files
+sonar.language=py
+# Use this for python projects:
+sonar.python.pylint=/usr/bin/pylint3
+sonar.python.pylint_config=pylintrc
+sonar.python.pylint.reportPath=pylint-report.txt
+# Ignore solar rules S107 S3776
+sonar.issue.ignore.multicriteria=e1,e2
+sonar.issue.ignore.multicriteria.e1.ruleKey=python:S107
```
Comment (dupremathieu) [inline sonar-project.properties:17]: Do we really want to ignore these rules?

**Review by dupremathieu — CHANGES_REQUESTED** [review]

File: sonar-project.properties (hunk 3 above)
Comment (eroussy) [reply sonar-project.properties:17]: It is done by project template

File: .cqfd/docker/Dockerfile (hunk 1 above)
Comment (eroussy) [reply .cqfd/docker/Dockerfile]: done

**Review by dupremathieu — APPROVED** [review]

**Pull request merged**